index.d.ts

```diff
@@ -1,184 +1,261 @@
-        getRowProps: () => any;
-        original: any;
-    }
+  export type SortingRules<D> = SortingRule<D>[]
 
-    export interface HeaderColumn<D, A extends keyof D = never> {
-        /**
-         * This string/function is used to build the data model for your column.
-         */
-        accessor: A | ((originalRow: D) => string);
-        Header?: string | ((props: TableInstance<D>) => ReactNode);
-        Filter?: string | ((props: TableInstance<D>) => ReactNode);
-        Cell?: string | ((cell: Cell<D>) => ReactNode);
-
-        /**
-         * This is the unique ID for the column. It is used by reference in things like sorting, grouping, filtering etc.
-         */
-        id?: string | number;
-        minWidth?: string | number;
-        maxWidth?: string | number;
-        width?: string | number;
-        canSortBy?: boolean;
-        sortByFn?: (a: any, b: any, desc: boolean) => 0 | 1 | -1;
-        defaultSortDesc?: boolean;
-    }
+  export type Filters<D> = {
+    [key: keyof D]: string
+  }
 
-    export interface Column<D, A extends keyof D = never> extends HeaderColumn<D, A> {
-        id: string | number;
-    }
+  export interface Cell<D = any> extends TableInstance<D> {
+    cell: { value: any }
+    column: Column<D>
+    row: Row<D>
+    render: (type: 'Cell' | 'Aggregated', userProps?: any) => any
+    isGrouped?: boolean
+    isAggregated?: boolean
+    isRepeatedValue?: boolean
+    getCellProps: () => any
+  }
 
-    export type Page<D> = Row<D>[];
+  export interface Row<D = any> {
+    index: number
+    cells: Cell<D>[]
+    getRowProps: (userProps?: any) => any
+    original: D
+    path: any[]
+    values: any[]
+    depth: number
+  }
 
-    export interface EnhancedColumn<D, A extends keyof D = never> extends Column<D, A> {
-        render: (type: string) => any;
-        getHeaderProps: (userProps?: any) => any;
-        getSortByToggleProps: (userProps?: any) => any;
-        sorted: boolean;
-        sortedDesc: boolean;
-        sortedIndex: number;
-    }
+  export interface UseExpandedRow<D = any> {
+    subRows?: D[]
+    groupByID?: string | number
+    toggleExpanded?: () => any
+    isExpanded?: boolean
+    isAggregated?: boolean
+  }
 
-    export type HeaderGroup<D, A extends keyof D = never> = {
-        headers: EnhancedColumn<D, A>[];
-        getRowProps: (userProps?: any) => any;
-    };
-
-    export interface Hooks<D> {
-        beforeRender: [];
-        columns: [];
-        headerGroups: [];
-        headers: [];
-        rows: Row<D>[];
-        row: [];
-        renderableRows: [];
-        getTableProps: [];
-        getRowProps: [];
-        getHeaderRowProps: [];
-        getHeaderProps: [];
-        getCellProps: [];
-    }
+  export interface HeaderColumn<D> {
+    /**
+     * This string/function is used to build the data model for your column.
+     */
+    accessor: string | ((originalRow: D) => string)
+    Header?: ReactText | ((props: TableInstance<D>) => ReactNode)
```

Comment:
`ReactText` is `string | number`. 

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/239f52df4b9d258b3ed6862499b0f369ce71ffb7/types/react/index.d.ts#L186-L191

IMO if the function can return a `ReactNode` then we should be able to put a `ReactNode` in without using the function.

Reply:
This was out of date.  in https://github.com/tannerlinsley/react-table/commit/0a03ea3a7521d62e34c6df9dba33e0aacb35bf41 the behavior changed and allows JSX again - so yes, ReactNode is now the right type.

It actually used to be this way, then react-table stopped supporting it, now it does again.  This is better :)